src/pages/taskQueue/index.tsx

```diff
@@ -1,120 +1,137 @@
+import { DISTRO_TASK_QUEUE, TASK_QUEUE_DISTROS } from "gql/queries";
 import { usePageTitle } from "hooks";
-import { DistroOption } from "pages/taskQueue/DistroOption";
-import { TaskQueueTable } from "pages/taskQueue/TaskQueueTable";
+import { useQueryParam } from "hooks/useQueryParam";
+import { QueryParams } from "types/task";
+import { DistroOption } from "./DistroOption";
+import TaskQueueTable from "./TaskQueueTable";
 
 const TaskQueue = () => {
   const taskQueueAnalytics = useTaskQueueAnalytics();
 
-  const { distro, taskId } = useParams<{ distro: string; taskId?: string }>();
+  const { distro } = useParams<{ distro: string }>();
+  const [taskId] = useQueryParam<string | undefined>(
+    QueryParams.TaskId,
+    undefined,
+  );
   const navigate = useNavigate();
-
-  const [selectedDistro, setSelectedDistro] = useState(null);
+  const [selectedDistro, setSelectedDistro] = useState<
+    TaskQueueDistro | undefined
+  >(undefined);
+  const dispatchToast = useToastContext();
   usePageTitle(`Task Queue - ${distro}`);
-  const { data: distrosData } = useQuery<
+  const { data: distrosData, loading: loadingDistrosData } = useQuery<
     TaskQueueDistrosQuery,
     TaskQueueDistrosQueryVariables
-  >(TASK_QUEUE_DISTROS, { fetchPolicy: "cache-and-network" });
-
-  const distros = useMemo(
-    () => distrosData?.taskQueueDistros ?? [],
-    [distrosData],
-  );
-  const firstDistroInList = distros[0]?.id;
-
-  // SET DEFAULT DISTRO AFTER DISTROS HAVE BEEN OBTAINED
-  useEffect(() => {
-    if (distros.length) {
+  >(TASK_QUEUE_DISTROS, {
+    fetchPolicy: "cache-and-network",
+    onCompleted: (data) => {
+      const { taskQueueDistros } = data;
+      const firstDistroInList = taskQueueDistros[0]?.id;
       const defaultDistro = distro ?? firstDistroInList;
-      setSelectedDistro(distros.find((d) => d.id === defaultDistro));
-      navigate(getTaskQueueRoute(defaultDistro, taskId), { replace: true });
-    }
-  }, [firstDistroInList, distro, navigate, taskId, distros]);
+      setSelectedDistro(taskQueueDistros.find((d) => d.id === defaultDistro));
+      if (distro === undefined) {
+        navigate(getTaskQueueRoute(defaultDistro, taskId));
+      }
+    },
+    onError: (err) => {
+      dispatchToast.error(`There was an error loading distros: ${err.message}`);
+    },
+  });
 
-  const onChangeDistroSelection = (val: { id: string }) => {
+  const { data: taskQueueItemsData, loading } = useQuery<
+    DistroTaskQueueQuery,
+    DistroTaskQueueQueryVariables
+  >(DISTRO_TASK_QUEUE, {
+    fetchPolicy: "cache-and-network",
+    variables: { distroId: distro },
+    skip: !distro,
+    onError: (err) => {
+      dispatchToast.error(
+        `There was an error loading task queue: ${err.message}`,
+      );
+    },
+  });
+
+  const onChangeDistroSelection = (val: TaskQueueDistro) => {
     taskQueueAnalytics.sendEvent({ name: "Select Distro", distro: val.id });
+    navigate(getTaskQueueRoute(val.id, taskId));
```

Comment:
I think maybe you don't want to carry over the taskID, since it'll be a different distro